classifier/systemVendor.go

```diff
@@ -0,0 +1,47 @@
+package classifier
+
+import (
+	bmh "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+
+	hwcc "github.com/metal3-io/hardware-classification-controller/api/v1alpha1"
+)
+
+func checkSystemVendor(profile *hwcc.HardwareClassification, host *bmh.BareMetalHost) bool {
+	systemVendorDetails := profile.Spec.HardwareCharacteristics.SystemVendor
+	if systemVendorDetails == nil {
+		return true
+	}
+
```

Comment:
The same comment about handling empty strings applies to this function.

Reply:
Done.